### PR TITLE
Spoonacular

### PR DESCRIPTION
**Pull request by aahmadyar123**
Merge API Integration Branch

### CODE REVIEW

File: backend/backend.js [hunk 1]
```diff
@@ -382,7 +385,7 @@ app.delete("/ingredients/:id", async (req, res) => {
     }
   } catch (error) {
     console.log(error);
-    res.status(500).send("Internal Server Error.");
+    res.status(501).send("Internal Server Error.");
```
Comment (luisdavidgarcia) [inline backend/backend.js:388]: I think this should've been kept as 500, I though 501 was for a non-implmented function [link](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#server_error_responses)

**Review by Westluu — COMMENTED** [review]
LGTM

File: backend/backend.js
Comment (Westluu) [inline backend/backend.js:1]: LGTM

File: backend/backend.js (hunk 1 above)
Comment (luisdavidgarcia) [reply backend/backend.js:388]: Otherwise Looks good to me

**Review by junseo-lee-git — APPROVED** [review]
Looks good

**Review by killian-brait — APPROVED** [review]
LGTM, we need to add the code to put the Spoonacular recipes into our database.

**Review by Westluu — APPROVED** [review]
The parsing functions look goods, good job adding the API request.

**Pull request merged**